docs/pages/experiments/base/menu.tsx

```diff
@@ -0,0 +1,160 @@
+  overflow: auto;
+  outline: 0px;
+  background: ${theme.palette.mode === 'dark' ? grey[900] : '#fff'};
+  border: 1px solid ${theme.palette.mode === 'dark' ? grey[700] : grey[200]};
+  color: ${theme.palette.mode === 'dark' ? grey[300] : grey[900]};
+  box-shadow: 0px 4px 30px ${theme.palette.mode === 'dark' ? grey[900] : grey[200]};
+  `,
+);
+
+const StyledMenuItem = styled(MenuItemUnstyled)(
+  ({ theme }) => `
+  list-style: none;
+  padding: 8px;
+  border-radius: 8px;
+  cursor: default;
+  user-select: none;
+
+  &:last-of-type {
+    border-bottom: none;
+  }
+
+  &.${menuItemUnstyledClasses.focusVisible} {
+    outline: 3px solid ${theme.palette.mode === 'dark' ? blue[600] : blue[200]};
+    background-color: ${theme.palette.mode === 'dark' ? grey[800] : grey[100]};
+    color: ${theme.palette.mode === 'dark' ? grey[300] : grey[900]};
+  }
+
+  &.${menuItemUnstyledClasses.disabled} {
+    color: ${theme.palette.mode === 'dark' ? grey[700] : grey[400]};
+  }
+
+  &:hover:not(.${menuItemUnstyledClasses.disabled}) {
+    background-color: ${theme.palette.mode === 'dark' ? grey[800] : grey[100]};
+    color: ${theme.palette.mode === 'dark' ? grey[300] : grey[900]};
+  }
+  `,
+);
+
+const MenuButton = styled('button')(
+  ({ theme }) => `
+  font-family: IBM Plex Sans, sans-serif;
+  font-size: 0.875rem;
+  box-sizing: border-box;
+  min-height: calc(1.5em + 22px);
+  border-radius: 12px;
+  padding: 12px 16px;
+  line-height: 1.5;
+  background: ${theme.palette.mode === 'dark' ? grey[900] : '#fff'};
+  border: 1px solid ${theme.palette.mode === 'dark' ? grey[700] : grey[200]};
+  color: ${theme.palette.mode === 'dark' ? grey[300] : grey[900]};
+
+  transition-property: all;
+  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
+  transition-duration: 120ms;
+
+  &:hover {
+    background: ${theme.palette.mode === 'dark' ? grey[800] : grey[50]};
+    border-color: ${theme.palette.mode === 'dark' ? grey[600] : grey[300]};
+  }
+
+  &:focus {
+    border-color: ${blue[400]};
+    outline: 3px solid ${theme.palette.mode === 'dark' ? blue[500] : blue[200]};
+  }
+  `,
+);
+
+const Popper = styled(PopperUnstyled)`
+  z-index: 1;
+`;
+
+const Page = styled('div')(`
+  max-width: 800px;
+  min-height: calc(100vh - 40px);
+  box-sizing: border-box;
+  margin: 20px auto;
+  padding: 20px;
+  background: ${grey[100]};
+  border-radius: 4px;
+`);
+
+export default function UnstyledMenuIntroduction() {
+  const [isOpen, setOpen] = React.useState(false);
+
+  const close = () => {
+    setOpen(false);
+  };
+
+  const handleOpenChange = (event: React.SyntheticEvent<HTMLButtonElement>, open: boolean) => {
+    setOpen(open);
+  };
+
+  const createHandleMenuClick = (menuItem: string) => {
+    return () => {
+      // eslint-disable-next-line no-console
+      console.log(`Clicked on ${menuItem}`);
+      close();
+    };
+  };
+
+  return (
+    <Page>
+      <MenuButtonUnstyled
+        slots={{ root: MenuButton }}
+        label="My account"
```

Comment:
The use of slots feels strange. Aren't the slot means to replace existing implementation for customization use cases? In this case, I would imagine that what's rendered as a menu trigger will most of the time be "userland". So I think that someone like this:
```suggestion
      <MenuButtonUnstyled
        trigger={<MenuButton>My account</MenuButton>}
```

would make more sense. Also notice that it's an element, not a component in this proposal. Why? For the extra flexibility needed when dealing with userland APIs. By "userland", I mean API that is used by the product teams in the normal course of building a product using a design system. This is to be distinguished from the API used by the product infrastructure team, the one building the design system with MUI Base.

Actually, in the demo, I think that this terminology would be clearer like this, to signal it can be any button:
```suggestion
      <MenuButtonUnstyled
        trigger={<MenuButton>My account</MenuButton>}
```

Reply:
> I would imagine that what's rendered as a menu trigger will most of the time be "userland"

Why do you think developers will want to use something different than `button` by default? Do you think Menu is different in this regard than, let's say, a Select?

Reply:
> Why do you think developers will want to use something different than button by default? Do you think Menu is different in this regard than, let's say, a Select?

I'm more thinking about the customization cases, between:

```jsx
      <MenuButtonUnstyled
        slots={{ root: MenuButton }}
        label="My account"
        variant="outlined"
        size="small"
     >
```

and:

```jsx
      <MenuButtonUnstyled
        trigger={
          <Button variant="outlined" size="small">
            My account
          </Button>
        }
      >
```

the latter seems clearer.